File: remoteconfig/src/main/java/me/giacoppo/remoteconfig/exceptions/HttpException.java
package me.giacoppo.remoteconfig.exceptions;

import java.io.IOException;

/**
 * Wraps the http code and server message in case of not successful response
 */
public final class HttpException extends IOException {
    private final int httpCode;

    public HttpException(int httpCode) {
        super("[http code: " + httpCode + "]");
        this.httpCode = httpCode;
    }

    public HttpException(int httpCode, String message) {
        super("[http code: " + httpCode + "] " + message);
        this.httpCode = httpCode;
    }

    public int getHttpCode() {
        return httpCode;
    }
}

File: remoteconfig/src/main/kotlin/me/giacoppo/remoteconfig/exceptions/HttpException.kt
package me.giacoppo.remoteconfig.exceptions

import java.io.IOException

/**
 * Wraps the http code and server message in case of not successful response
 */
class HttpException : IOException {
    val httpCode: Int

    constructor(httpCode: Int) : super("[http code: $httpCode]") {
        this.httpCode = httpCode
    }

    constructor(httpCode: Int, message: String) : super("[http code: $httpCode] $message") {
        this.httpCode = httpCode
    }
}
